lib/eslint.js

```diff
@@ -998,6 +998,11 @@ module.exports = (function() {
             i,
             len;
 
+        // Special Node.js scope means we need to start one level deeper
+        if (scope.type === "global" && currentConfig.ecmaFeatures.globalReturn) {
```

Comment:
Not really (I do not clearly understand how these scopes are working :( ), I just noticed we got the same behavior when we have `globalReturn` or `modules` to `true`: we need to check `scope.childScopes[0]` to get all the variables.

Test to reproduce the issue:

```js
var code = "var a = 1, b = 2;";

eslint.reset();
eslint.on("Program:exit", function() {
	var globalScope = eslint.getScope(),
	childScope = globalScope.childScopes[0];

	eslint.markVariableAsUsed("a");

	assert.isTrue(getVariable(childScope, "a").eslintUsed);
	assert.isUndefined(getVariable(childScope, "b").eslintUsed);
});

eslint.verify(code, { ecmaFeatures: { modules: true }}, filename, true);
```